src/py/flwr/server/app.py

```diff
@@ -375,7 +381,12 @@ def run_server() -> None:
     )
 
     # Block
-    driver_server.wait_for_termination()
+    while True:
+        if bckg_threads:
+            for thread in bckg_threads:
+                if not thread.is_alive():
+                    sys.exit(1)
+        driver_server.wait_for_termination(timeout=1)
```

Comment:
@tanertopal is this related to HTTPS, or is it a general improvement? If it's not related to HTTPS, let's move it to a separate PR

Reply:
@danieljanes I removed this.